src/Symfony/Component/Serializer/Tests/Encoder/CsvEncoderTest.php

```diff
@@ -36,15 +36,45 @@ public function testTrueFalseValues()
             'int' => 2,
             'false' => false,
             'true' => true,
+            'int_one' => 1,
+            'string_one' => '1',
         ];
 
         // Check that true and false are appropriately handled
-        $this->assertEquals(<<<'CSV'
-string,int,false,true
-foo,2,0,1
+        $this->assertSame($csv = <<<'CSV'
+string,int,false,true,int_one,string_one
+foo,2,0,1,1,1
+
+CSV
+        , $this->encoder->encode($data, 'csv'));
+
+        $this->assertSame($result = [
+            'string' => 'foo',
+            'int' => '2',
+            'false' => '0',
+            'true' => '1',
+            'int_one' => '1',
+            'string_one' => '1',
+        ], $this->encoder->decode($csv, 'csv'));
+    }
+
+    public function testDoubleQuotesAndSlashes()
+    {
+        $this->assertSame($csv = <<<'CSV'
+0,1,2,3,4,5
+,"""","foo""","\"","\","foo\"
 
 CSV
-    , $this->encoder->encode($data, 'csv'));
+        , $this->encoder->encode($data = ['', '"', 'foo"', '\\"', '\\', 'foo\\'], 'csv'));
+
+        $this->assertSame($result = ['', '"', 'foo"', '\\"', '\\",foo\\"'], $this->encoder->decode($csv, 'csv'));
```

Comment:
we lose 1 element (expected is 6, got 5) by decoding a previously encoded result